lib/ex_doc/retriever.ex

```diff
@@ -158,16 +158,18 @@ defmodule ExDoc.Retriever do
   defp generate_node(module, type, config) do
     source_url  = config.source_url_pattern
     source_path = source_path(module, config)
+    config = %{source_url: source_url, source_path: source_path}
```

Comment:
Since we already have a variable named `config`, this can be confusing. What about calling this `source` with fields `url` and `path`?